xbmc/music/MusicDatabase.cpp

```diff
@@ -11177,13 +11534,29 @@ bool CMusicDatabase::GetFilter(CDbUrl &musicUrl, Filter &filter, SortDescription
     option = options.find("compilation");
     if (option != options.end())
       filter.AppendWhere(PrepareSQL("songview.bCompilation = %i", option->second.asBoolean() ? 1 : 0));
+    
+    option = options.find("boxset");
+    if (option != options.end())
+      filter.AppendWhere(PrepareSQL("EXISTS (SELECT 1 FROM album "
+        "WHERE album.idAlbum = songview.idAlbum AND albumview.bBoxedSet = %i)", option->second.asBoolean() ? 1 : 0));
```

Comment:
"WHERE album.idAlbum = songview.idAlbum AND album.bBoxedSet = %i)", option->second.asBoolean() ? 1 : 0));